### PR TITLE
Fix ledger sequential access errors

### PR DESCRIPTION
**Pull request by nddeluca**
- Add a `LedgerMock` class for testing.
  - It uses `setTimeout` to mimic non-blocking IO operations a real ledger has.
  - Non-sequential access will throw an error.
  - It's a bit barebones, but we can improve it over time with an actual test mnemonic to make signing work against a real test chain in CI, dev, etc.
- Adds unit tests for the `CosmosLedger` class that covers the queued operation behavior.
  - `Promise.all` is used to simulate parallel usage of the ledger device across react components
- Manually tested the code against a real device
  - On `master` branch, clicking the tabs fast will throw a sequential access error
  - On this branch, no errors are thrown.
- Add JSDoc
  - Run with `npm run doc`
  - View with `open doc/index.html`
  - Added `CosmosLedger` documentation

TODO before merge:
- [x] add tests to ensure errors go to correct caller (the reject promise path)
- [x] add doc comments to LedgerMock
- [x] update pubkey and signature LedgerMock methods to return correct data type
- [x] add code coverage npm script

### CODE REVIEW

File: src/features/ledger/ledgerMock.js [hunk 1]
```diff
@@ -0,0 +1,61 @@
+export const TEST_PUB_KEY = 'some public key';
+export const TEST_COSMOS_ADDRESS = 'some kava address';
+
+export class LedgerMock {
```
Comment (nddeluca) [inline src/features/ledger/ledgerMock.js:4]: We can turn this into a full fledged wallet and add a development flag in App.js in the future, so we can develop and run integration tests against a local running chain or testnet without an actual ledger device.

File: src/features/ledger/ledger.js [hunk 2]
```diff
@@ -1,27 +1,114 @@
...
+   * Queues the ledger operation returning a Promise that resolves
+   * with the result
+   *
+   * @param {Promise} ledgerFn A async ledger function
+   * @param {...*} ...args Arguments for the ledger function
+   * @returns {Promise} A Promise that resolves with the result of the ledger operation
+   * @private
+   */
+  _queue_op(ledgerFn, ...args) {
+    const op = new Promise((resolve, reject) => {
```
Comment (nddeluca) [inline src/features/ledger/ledger.js:73]: I started with an `async/await` approach, but found that using plain Promise's had a simpler implementation.

**Comment by denalimarsh** [comment]
Generally, we use 'R4R' and 'WIP' tags to indicate if a PR is ready for review or still a work in progress. I added the tag types to this repo and applied the WIP tag to this PR as it looks like there are still some TODOs before merge.

**Comment by nddeluca** [comment]
Added code coverage reporting.

Can run with `npm run coverage` and view with `open ./coverage/lcov-report/index.html`.

Screens of html report:
Screen Shot 2020-07-02 at 10 02 30 AM
Screen Shot 2020-07-02 at 10 00 52 AM

**Pull request merged**